src/main/java/com/rtb/projectmanagementtool/project/ProjectData.java

```diff
@@ -0,0 +1,204 @@
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.HashSet;
+
+enum User {
+  CREATOR,
+  ADMIN,
+  REGULAR;
+}
+
+public class ProjectData {
+  /**
+   * constant variables used to specify the names of each property of the Project entity. Used to
+   * minimize redundancy and also make renaming properties easy if needed
+   */
+  private final String PROPERTY_NAME = "name";
+
+  private final String PROPERTY_DESCRIPTION = "description";
+  private final String PROPERTY_USER_IDS = "userIds";
+  private final String PROPERTY_TASK_IDS = "taskIds";
+
+  private DatastoreService datastore;
+
+  private long id;
+  private String name;
+  private String description;
+  private HashMap<Long, User> userIds;
+  private HashSet<Long> taskIds;
+
+  /**
+   * Class constructor with the minimum requirements for creating a project.
+   *
+   * @param name the name of the project
+   * @param description the description of the project
+   * @param creatorId the id of the user who created the project
+   */
+  public ProjectData(String name, String description, long creatorId) {
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    this.id = datastore.put(new Entity("Project")).getId();
+    this.name = name;
+    this.description = description;
+
+    this.userIds = new HashMap<Long, User>();
+    this.userIds.put(creatorId, User.CREATOR);
+
+    this.taskIds = new HashSet<Long>();
+    toEntity();
+  }
+
+  /**
+   * Class constructor to create Project from existing datastore entity
+   *
+   * @param entity the datastore entity to parse object from
+   */
+  public ProjectData(Entity entity) {
+    datastore = DatastoreServiceFactory.getDatastoreService();
+
+    this.id = (Long) entity.getKey().getId();
+    this.name = (String) entity.getProperty(PROPERTY_NAME);
+    this.description = (String) entity.getProperty(PROPERTY_DESCRIPTION);
+
+    Gson gson = new Gson();
+
+    // sets the type of containers to store parsed JSON in
+    Type mapType = new TypeToken<HashMap<Long, User>>() {}.getType();
+    Type setType = new TypeToken<HashSet<Long>>() {}.getType();
+
+    this.userIds = gson.fromJson((String) entity.getProperty(PROPERTY_USER_IDS), mapType);
+    this.taskIds = gson.fromJson((String) entity.getProperty(PROPERTY_TASK_IDS), setType);
+  }
+
+  /** Saves the object as a datastore entity. */
+  private void toEntity() {
+    Gson gson = new Gson();
+    try {
+      Key entityKey = KeyFactory.createKey("Project", this.id);
+      Entity entity = datastore.get(entityKey);
+
+      entity.setProperty(PROPERTY_NAME, this.name);
+      entity.setProperty(PROPERTY_DESCRIPTION, this.description);
+      entity.setProperty(PROPERTY_USER_IDS, gson.toJson(this.userIds));
```

Comment:
This is an interesting idea.  While storing as json would allow you to store and later on read more formatted data, this will also require parsing later on.  This might work out well (maybe) if we're storing child objects that has multiple fields.  In this specific scenario though, since we're only storing a list of IDs, try to see if there's a better alternative.

Also keep in mind one caveat with storing as json is you wouldn't be able to query against this field.  For example if project userIDs is stored as json, it'll be hard to query for "all projects user X owns"